impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -93,7 +89,7 @@ public void write(XmlWriterRequest<PluginDescriptor> request) throws XmlWriterEx
                 new PluginDescriptorStaxWriter().write(outputStream, content);
             } else {
                 try (OutputStream os = Files.newOutputStream(path)) {
-                    new PluginDescriptorStaxWriter().write(outputStream, content);
+                    new PluginDescriptorStaxWriter().write(os, content);
```

Comment:
That line does look like a bug

Reply:
yes already done:
- https://github.com/apache/maven/pull/2303

This PR is follow up. Thx